packages/core/src/indexing-store/utils/encoding.ts

```diff
@@ -208,6 +220,8 @@ function decodeValue(
       return JSON.parse(value as string);
     }
     return value as UserColumn;
+  } else if (isJSONColumn(column)) {
+    return encoding === "postgres" ? value : JSON.parse(value as string);
```

Comment:
Initially thought we might want a try/catch here as well, but I suppose there's a very low risk of an error with this `JSON.parse`, because we already stringified it and managed to insert + select it out of the database.